File: Symfony/src/BGR/Serrano/ProductoBundle/Entity/Mix.php
<?php

namespace BGR\Serrano\ProductoBundle\Entity;

use Doctrine\ORM\Mapping as ORM;
use JMS\Serializer\Annotation\Type;

/**
 * Mix
 *
 * @ORM\Table()
 * @ORM\Entity(repositoryClass="BGR\Serrano\ProductoBundle\Entity\MixRepository")
 */
class Mix
{
    /**
     * @var integer
     *
     * @ORM\Column(name="id", type="integer")
     * @ORM\Id
     * @ORM\GeneratedValue(strategy="AUTO")
     */
    private $id;

    /**
     * @var string
     *
     * @ORM\Column(name="nombre", type="string", length=50)
     */
    private $nombre;

    /**
     * @var string
     *
     * @ORM\Column(name="descripcion", type="string", length=200)
     */
    private $descripcion;

    /**
     * @var \DateTime
     * @Type("DateTime<'Y-m-d'>")
     * @ORM\Column(name="vencimiento", type="date")
     */
    private $vencimiento;
        
    /**
     * @ORM\ManyToOne(targetEntity="UnidadDeMedida")
     * @ORM\JoinColumn(name="unidad_de_medida_id", referencedColumnName="id")
     * @Type("BGR\Serrano\ProductoBundle\Entity\UnidadDeMedida")
     */
    protected $unidad_de_medida;
    
    /**
     * @ORM\ManyToOne(targetEntity="Categoria") 
     * @ORM\JoinColumn(name="categoria_id", referencedColumnName="id")
     * @Type("BGR\Serrano\ProductoBundle\Entity\Categoria")
     */
    protected $categoria;
	
	/**
	 *
	 * @return the integer
	 */
	public function getId() {
		return $this->id;
	}
	
	/**
	 *
	 * @param
	 *        	$id
	 */
	public function setId($id) {
		$this->id = $id;
		return $this;
	}
	
	/**
	 *
	 * @return the string
	 */
	public function getNombre() {
		return $this->nombre;
	}
	
	/**
	 *
	 * @param string $nombre        	
	 */
	public function setNombre($nombre) {
		$this->nombre = $nombre;
		return $this;
	}
	
	/**
	 *
	 * @return the string
	 */
	public function getDescripcion() {
		return $this->descripcion;
	}
	
	/**
	 *
	 * @param string $descripcion        	
	 */
	public function setDescripcion($descripcion) {
		$this->descripcion = $descripcion;
		return $this;
	}
	
	/**
	 *
	 * @return the DateTime
	 */
	public function getVencimiento() {
		return $this->vencimiento;
	}
	
	/**
	 *
	 * @param  $vencimiento        	
	 */
	public function setVencimiento($vencimiento) {
		$this->vencimiento = $vencimiento;
		return $this;
	}
	
	/**
	 *
	 * @return the unknown_type
	 */
	public function getUnidadDeMedida() {
		return $this->unidad_de_medida;
	}
	
	/**
	 *
	 * @param unknown_type $unidad_de_medida        	
	 */
	public function setUnidadDeMedida($unidad_de_medida) {
		$this->unidad_de_medida = $unidad_de_medida;
		return $this;
	}
	
	/**
	 *
	 * @return the unknown_type
	 */
	public function getCategoria() {
		return $this->categoria;
	}
	
	/**
	 *
	 * @param unknown_type $categoria        	
	 */
	public function setCategoria($categoria) {
		$this->categoria = $categoria;
		return $this;
	}
	
    
}

File: BGR/BGR-Data/src/main/java/ar/com/bgr/serrano/model/Mix.java
package ar.com.bgr.serrano.model;

import java.util.Date;

import javax.persistence.Column;
import javax.persistence.Entity;
import javax.persistence.FetchType;
import javax.persistence.GeneratedValue;
import javax.persistence.Id;
import javax.persistence.JoinColumn;
import javax.persistence.ManyToOne;
import javax.persistence.Table;

@Entity(name="Mix")
@Table(name="Mix")
public class Mix {
	
	@Id
    @GeneratedValue
    private Integer id;

    @Column(name="name")  
    private String name;

    @Column(name="descripcion")  
    private String descripcion;

    @Column(name="vencimiento")  
    private Date vencimiento;
        
    @ManyToOne
    @JoinColumn(name="unidad_de_medida_id")
    private UnidadDeMedida unidad_de_medida;

	@ManyToOne(fetch = FetchType.EAGER)	
	@JoinColumn(name = "categoria_id", nullable = false)
    private Categoria categoria;

	/**
	 * @return the id
	 */
	public Integer getId() {
		return id;
	}

	/**
	 * @param id the id to set
	 */
	public void setId(Integer id) {
		this.id = id;
	}

	/**
	 * @return the name
	 */
	public String getName() {
		return name;
	}

	/**
	 * @param name the name to set
	 */
	public void setName(String name) {
		this.name = name;
	}

	/**
	 * @return the descripcion
	 */
	public String getDescripcion() {
		return descripcion;
	}

	/**
	 * @param descripcion the descripcion to set
	 */
	public void setDescripcion(String descripcion) {
		this.descripcion = descripcion;
	}

	/**
	 * @return the vencimiento
	 */
	public Date getVencimiento() {
		return vencimiento;
	}

	/**
	 * @param vencimiento the vencimiento to set
	 */
	public void setVencimiento(Date vencimiento) {
		this.vencimiento = vencimiento;
	}

	/**
	 * @return the unidad_de_medida
	 */
	public UnidadDeMedida getUnidad_de_medida() {
		return unidad_de_medida;
	}

	/**
	 * @param unidad_de_medida the unidad_de_medida to set
	 */
	public void setUnidad_de_medida(UnidadDeMedida unidad_de_medida) {
		this.unidad_de_medida = unidad_de_medida;
	}

	/**
	 * @return the categoria
	 */
	public Categoria getCategoria() {
		return categoria;
	}

	/**
	 * @param categoria the categoria to set
	 */
	public void setCategoria(Categoria categoria) {
		this.categoria = categoria;
	}
    
    
    
}
